### ISSUE TITLE
Integration with DSpace and authentication

### ISSUE THREAD

**Issue by nemobis** [dissemin/dissemin#797]
(I understand that [OpenIng](http://www.opening-projekt.de/) will [work on a DSpace integration](https://github.com/dissemin/dissemin/projects/5) at some point, so this will become relevant. If not, feel free to close!)

The [presentation about EasyDeposit2](https://www.conftool.net/or2019/index.php/Paper-P5C-117Zhang.pdf?page=downloadPaper&filename=Paper-P5C-117Zhang.pdf&form_id=117) by the University of Oregon at the [green OA](https://www.conftool.net/or2019/index.php?page=browseSessions&form_session=375) session of [OR2019](https://or2019.blogs.uni-hamburg.de/) showed how https://github.com/osulp/Easydeposit2 is used to allow researchers to upload a PDF to a repository record in reaction to an email without needing to login manually.

Some of the logic might turn out to be useful if we want to deposit on the authors' behalf on DSpace and/or generate email notifications with access tokens. I think these were the crucial points: https://github.com/osulp/Scholars-Archive/issues/1603 https://github.com/osulp/Easydeposit2/blob/84d70186137e91ec274839caf952bcaaf11eb016/lib/osu_api/client.rb .

**Comment by beckstefan** [comment]
Thank you. This sounds indeed interesting.

Nevertheless, I doubt that this is a way to go for OpenIng, because sending emails undergoes some restrictions, i.e. legally you cannot just send someone automated emails without their agreement. Probably different in US.

**Comment by nemobis** [comment]
Note that the University of Oregon folks were writing to their own researchers.

**Comment by beckstefan** [comment]
I close this, since it won't be tackled by OpenIng.

> Note that the University of Oregon folks were writing to their own researchers.

This does, as far as I know, no difference in Germany.

**Comment by nemobis** [comment]
Are you closing this just to mark it as off scope for OpenING, or also 
because you think it's not worth pursuing an integration with DSpace 
which makes it easier for the user to authenticate to the target repository?

Stefan Beck, 14/01/20 12:10:
>     Note that the University of Oregon folks were writing to their own
>     researchers.
> 
> This does, as far as I know, no difference in Germany.

Why not? It would be pretty hard for the office of university A to make 
a user authenticated on the IR of university !A, I suppose.

Federico

**Comment by beckstefan** [comment]
Both, although I do not think it is not worth in general, but I don't see use cases in the near future. If you like the issue to be open, please feel free to open it again! :-)

The point is not about authentication, but sending automated or systematically e-mails. Unless the user does not agree to receive such e-mails, you are not allowed to send such e-mails.

**Comment by nemobis** [comment]
Stefan Beck, 15/01/20 15:25:
> Unless the user does not agree to receive such e-mails, you are not 
> allowed to send such e-mails.

This is not a correct representation of the law, but I understand it's 
your internal policy so for the sake of OpenING it doesn't matter. Other 
users may have less strict policies and therefore have a use for the 
feature.

**Comment by nemobis** [comment]
Reopening in light of Oregon State University's https://doi.org/10.3390/publications8020028 : email tokens are a valid authentication mechanism used by countless websites (in EU too). Email collection is a separate matter and can be handled by whoever runs the Dissemin instance, which may not be in EU.

**Comment by wetneb** [comment]
I agree that email tokens would be a great benefit for email campaigns.
